### PR TITLE
Bug 1188797 - [System] Implement alwaysLowered window.open() feature …

### PR DESCRIPTION
**Pull request by arcturus**
…in Gaia r=etienne

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=32fc6c87711b72e01b0c60efc20b45a03a200c39

File: apps/system/test/unit/child_window_factory_test.js [hunk 1]
```diff
@@ -404,6 +406,48 @@ suite('system/ChildWindowFactory', function() {
...
+      var spyPopupWindow = this.sinon.spy(window, 'PopupWindow');
+      var cwf = new ChildWindowFactory(app1);
+      var spyCreatePopupWindow = this.sinon.spy(cwf, 'createPopupWindow');
+      cwf.handleEvent(new CustomEvent('mozbrowseropenwindow',
+        {
+          detail: fakeWindowOpenHiddenPopup
+        }));
+      assert.isTrue(spyCreatePopupWindow.calledOnce);
+      assert.isTrue(spyPopupWindow.getCall(0).args[0].stayBackground);
+    });
```
Comment (etiennesegonzac) [inline apps/system/test/unit/child_window_factory_test.js]: let's merge those 2 tests since there's a lot of boilerplate.

Comment (etiennesegonzac) [inline apps/system/test/unit/child_window_factory_test.js]: and then extract those tests to run them once against `alwaysLowered` and once against `z-lock` :)
(or we can remove the `z-lock` support from the code, I'm fine with that too)

File: apps/system/js/popup_window.js [hunk 2]
```diff
@@ -28,6 +28,11 @@
       return;
     }
 
+    // Check if the popup is requested to be always lowered
+    if (configs.stayBackground) {
+      this.element.classList.add('alwaysLowered');
```
Comment (etiennesegonzac) [inline apps/system/js/popup_window.js]: we also want the `openAnimation` and `closeAnimation` to return "immediate" in this case

**Comment by etiennesegonzac** [comment]
We also need to update `AppTransitionController#_shouldFocusApp` [1] I think to skip those windows.
[1] https://github.com/mozilla-b2g/gaia/blob/6d91a3d13cc2a2f4a1f05e4db1076ca31517f928/apps/system/js/app_transition_controller.js#L284-L290

File: apps/system/test/unit/child_window_factory_test.js (hunk 1 above)
Comment (arcturus) [inline apps/system/test/unit/child_window_factory_test.js]: +1 to remove the z-lock support

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=103a25f6a7e09aaec64a5fd3edc60cb40c9b524f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2155f91cc9e848f580375248c5a48b966865b32b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2d42ff160d2c9f1a756e9fa9904172e6738f236c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9a726b89c4a95afc77c148de5b58865afc267f2a

**Pull request merged**